tests/test_isrFunctions.py

```diff
@@ -0,0 +1,402 @@
+
+        Expect number of mask pixels with SAT marked to be non-zero.
+        """
+        for interpolate in (True, False):
+            with self.subTest(interpolate=interpolate):
+                corrMaskedImage = ipIsr.saturationCorrection(self.mi, 200, 2.0,
+                                                             growFootprints=2, interpolate=interpolate,
+                                                             maskName='SAT')
+                numBit = maskCountPixels(corrMaskedImage, "SAT")
+                self.assertGreaterEqual(numBit, 0)
+
+    def test_trimToMatchCalibBBox(self):
+        r"""Trim to match raw data.
+
+        Expect bounding boxes to match.
+        """
+        darkExp = isrMock.DarkMock().run()
+        darkMi = darkExp.getMaskedImage()
+
+        nEdge = 2
+        newDark = darkMi[nEdge:-nEdge, nEdge:-nEdge, afwImage.LOCAL]
+        darkMi = newDark
+        newInput = ipIsr.trimToMatchCalibBBox(self.mi, darkMi)
+
+        self.assertEqual(newInput.getImage().getBBox(), newDark.getImage().getBBox())
+
+    def test_darkCorrection(self):
+        r"""Dark correction.
+
+        Expect round-trip application to be equal.
+        Expect RuntimeError if sizes are different.
+        """
+        darkExp = isrMock.DarkMock().run()
+        darkMi = darkExp.getMaskedImage()
+
+        image = self.mi.getImage().getArray()
+        ipIsr.darkCorrection(self.mi, darkMi, 1.0, 1.0, trimToFit=True)
+        ipIsr.darkCorrection(self.mi, darkMi, 1.0, 1.0, trimToFit=True, invert=True)
+
+        diff = image - self.mi.getImage().getArray()
+        assert (diff.sum() == 0)
+
+        newDark = darkMi[1:-1, 1:-1, afwImage.LOCAL]
+        darkMi = newDark
+
+        with self.assertRaises(RuntimeError):
+            ipIsr.darkCorrection(self.mi, darkMi, 1.0, 1.0, trimToFit=False)
+
+    def test_biasCorrection(self):
+        r"""Bias correction.
+
+        Expect smaller median image value after.
+        Expect RuntimeError is sizes are different.
+        """
+        biasExp = isrMock.BiasMock().run()
+        biasMi = biasExp.getMaskedImage()
+
+        ipIsr.biasCorrection(self.mi, biasMi, trimToFit=True)
+
+        assert True
+
+        newBias = biasMi[1:-1, 1:-1, afwImage.LOCAL]
+        biasMi = newBias
+
+        with self.assertRaises(RuntimeError):
+            ipIsr.biasCorrection(self.mi, biasMi, trimToFit=False)
+
+    def test_flatCorrection(self):
+        r"""Flat correction.
+
+        Expect round-trip application to be equal
+        Expect RuntimeError if sizes are different or if scaling type isn't known.
+        """
+        flatExp = isrMock.FlatMock().run()
+        flatMi = flatExp.getMaskedImage()
+
+        image = self.mi.getImage().getArray()
+        for scaling in ('USER', 'MEAN', 'MEDIAN', 'UNKNOWN'):
+            with self.subTest(scaling=scaling):
+                if scaling == 'UNKNOWN':
+                    with self.assertRaises(pexExcept.Exception):
```

Comment:
Which specific exception is this supposed to raise?

Reply:
This now raises a RuntimeError instead of leaving a generic exception.